### PR TITLE
Remove explicit typecast and replace with <>

### PR DESCRIPTION
**Pull request by mmoayyed**
So this:

```java
Map<String, Object> EMPTY_MAP = Collections.unmodifiableMap(new HashMap<String, Object>());
```

Becomes:

```java
Map<String, Object> EMPTY_MAP = Collections.unmodifiableMap(new HashMap<>());
```

### CODE REVIEW

File: cas-server-core-services/src/main/java/org/jasig/cas/authentication/principal/AbstractWebApplicationService.java [hunk 1]
```diff
@@ -21,7 +21,7 @@
 
     private static final long serialVersionUID = 610105280927740076L;
 
-    private static final Map<String, Object> EMPTY_MAP = Collections.unmodifiableMap(new HashMap<String, Object>());
+    private static final Map<String, Object> EMPTY_MAP = Collections.unmodifiableMap(new HashMap<>());
```
Comment (frett) [inline cas-server-core-services/src/main/java/org/jasig/cas/authentication/principal/AbstractWebApplicationService.java]: minor nit, we could probably just use `Collections.emptyMap()` or `ImmutableMap.of()` here.

**Comment by frett** [comment]
+1, just a minor nit

Since I've switched my own CAS deployment to target Java 8 I've started to notice quite a few places that I no longer need explicit generics. It's been nice since most of those places annoyed me that I had to be explicit for no apparent reason.

**Comment by mmoayyed** [comment]
Agreed. It's much leaner this way. Less noisy :) 

One of the things I really miss in the Java platform is extension methods like C# does. I imagine we could have cleaned up the codebase much more if we had something like that.

File: cas-server-core-services/src/main/java/org/jasig/cas/authentication/principal/AbstractWebApplicationService.java (hunk 1 above)
Comment (mmoayyed) [inline cas-server-core-services/src/main/java/org/jasig/cas/authentication/principal/AbstractWebApplicationService.java]: Sure. Will do.

**Pull request merged**